test/kafka-ingest-open-loop/mzcompose.py

```diff
@@ -0,0 +1,187 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import time
+
+import pg8000
+
+from materialize.mzcompose import Composition, WorkflowArgumentParser, say
+from materialize.mzcompose.services import (
+    Kafka,
+    Kgen,
+    Materialized,
+    Testdrive,
+    Zookeeper,
+)
+
+mz_persist = Materialized(
+    name="materialized_persist",
+    hostname="materialized",
+    workers=4,
+    timestamp_frequency="1s",
+    options="--persistent-user-tables --persistent-kafka-upsert-source --disable-persistent-system-tables-test",
+)
+
+mz_default = Materialized(workers=4, timestamp_frequency="1s")
+
+prerequisites = ["zookeeper", "kafka"]
+
+SERVICES = [
+    Zookeeper(),
+    Kafka(),
+    mz_default,
+    mz_persist,
+    # This instance of Testdrive is used to initialize the benchmark, and needs to have a
+    # known seed.
+    Testdrive(no_reset=True, seed=1),
+    Kgen(),
+]
+
+
+def query_materialize(
+    c: Composition,
+    mz_service: str,
+) -> int:
+    host_port = c.default_port(mz_service)
+    query = "SELECT * FROM num_records_ingested"
+
+    try:
+        conn = pg8000.connect(
+            host="localhost", port=host_port, user="materialize", timeout=1
+        )
+        conn.autocommit = True
+
+        cursor = conn.cursor()
+        cursor.execute("SELECT * FROM load_test_materialization_frontier")
```

Comment:
I think you should be able to replace most of this with `Composition.sql` if you change that method to return the cursor. (Or add a new method like `sql_cursor` that returns a ready-to-use cursor, and use it in `Composition.sql` and here.)

Reply:
done. added `sql_cursor()` in a separate commit.